src/tools/remote-test-server/src/main.rs

```diff
@@ -253,10 +283,17 @@ fn recv<B: BufRead>(dir: &Path, io: &mut B) -> PathBuf {
     let dst = dir.join(t!(str::from_utf8(&filename[..len])));
     let amt = read_u32(io) as u64;
     t!(io::copy(&mut io.take(amt), &mut t!(File::create(&dst))));
-    t!(fs::set_permissions(&dst, Permissions::from_mode(0o755)));
+    set_permissions(&dst);
     dst
 }
 
+#[cfg(not(windows))]
+fn set_permissions(path: &Path) {
+    t!(fs::set_permissions(&dst, Permissions::from_mode(0o755)));
```

Comment:
I think `dst` should be `path` here

Reply:
Argh, yes! Really gotta setup WSL2 or a VM to test this, will do that later today.

(Is there a way to quickly run the equivalent of cargo check for different targets that don't support building form the host?)

Reply:
You can use [cross](https://github.com/rust-embedded/cross/)
e.g.
```sh
cross check --target x86_64-unknown-linux-gnu
```

Reply:
Thanks, will check it out later today